### PR TITLE
Add RecurringApplicationCharge support

### PR DESCRIPTION
**Pull request by amwolff**
I have added RecurringApplicationCharge (https://help.shopify.com/api/reference/billing/recurringapplicationcharge) support to the API. I tried to achieve consistency with the rest of the lib and, of course, if you spot otherwise - I'll be happy to clear that out.

There was unfortunately a problem. If you take a look at the provided docs, you can see that the RecurringApplicationCharge Properties are not actually all the fields that Shopify could return in a response. This is particularly observed when creating a recurring app charge with a capped amount + terms. Therefore I had looked at all the provided responses and gathered all the possible (I think?) fields. This code is going to get battle-tested at my company, so if anything unforeseen occurs, I'm going to fix that immediately and make a PR.

### CODE REVIEW

File: recurringapplicationcharge.go [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+// RecurringApplicationChargeServiceOp handles communication with the
+// RecurringApplicationCharge related methods of the Shopify API.
+type RecurringApplicationChargeServiceOp struct {
+	client *Client
+}
+
+// RecurringApplicationCharge represents a Shopify RecurringApplicationCharge.
+type RecurringApplicationCharge struct {
+	APIClientID           int              `json:"api_client_id"`
+	ActivatedOn           *time.Time       `json:"activated_on "`
```
Comment (dlebech) [inline recurringapplicationcharge.go]: There's a space at the end of the property string `json:"activated_on "`

**Review by dlebech — CHANGES_REQUESTED** [review]
Thanks @amwolff :trophy: ! I spotted a very small typo that needs fixing, but otherwise this looks good to go :beers:

File: recurringapplicationcharge.go (hunk 1 above)
Comment (amwolff) [reply recurringapplicationcharge.go]: This is now fixed, thanks for spotting that typo, great eye!

**Comment by amwolff** [comment]
Thank your for reviewing the code. I have addressed the review with the latest commit.

**Review by dlebech — APPROVED** [review]

**Comment by dlebech** [comment]
Thanks @amwolff ! :1st_place_medal:

**Pull request merged**